### DISCUSSION TITLE
Error in plugin @nomiclabs/hardhat-etherscan: The contract verification failed.

### DISCUSSION THREAD

**Discussion by jlescalonap** [NomicFoundation/hardhat#3389] [Troubleshooting] [answered]
Hi guys, I'm getting the following error from trying to verify my smartcontract on Etherscan

`We tried verifying your contract terceiroTest without including any unrelated one, but it failed.
Trying again with the full solc input used to compile and deploy it.
This means that unrelated contracts may be displayed on Etherscan...

Successfully submitted source code for contract
contracts/terceiroTest.sol:terceiroTest at 0x7842394fFceF46b4ca161249ff45a781b56e90f9
for verification on the block explorer. Waiting for verification result...

**Error in plugin @nomiclabs/hardhat-etherscan: The contract verification failed.
Reason: Fail - Unable to verify**

For more info run Hardhat with --show-stack-traces`

**Comment by fvictorio** [comment, marked as answer]
99% of the time, when someone gets the "Unable to verify" error, is because one of two reasons:

- One of your constructor arguments is wrong
- There was a temporary problem in the Etherscan servers

So, triple-check your constructor arguments (if any), or just try again after a while.

**Reply by jlescalonap** [reply]
Thanks very much for the response, I've started over the proyect and it worked seems that the error was something on the setup.
